python/tempo/interpol.py

```diff
@@ -0,0 +1,322 @@
+import sys
+from datetime import datetime
+from typing import List, Tuple
+
+from pyspark.sql import SparkSession as spark
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import (
+    col,
+    first,
+    last,
+    lit,
+    to_timestamp,
+    udf,
+    unix_timestamp,
+    when,
+)
+from pyspark.sql.types import FloatType
+from pyspark.sql.window import Window
+
+from tempo import *
+
+# Interpolation fill options
+fill_options = ["zero", "null", "back", "forward", "linear"]
+supported_target_col_types = ["double", "float"]
+
+
+class Interpolation:
+    def __init__(self):
+        """
+        Constructor
+
+        :linear_udf register linear calculation UDF
+        """
+        self.linear_udf = udf(Interpolation.__calc_linear, FloatType())
```

Comment:
@guanjieshen, is there any reason this needs to be a Python UDF vs pandas UDF? We should test this on a billion-record dataset to measure performance and I'm happy to help with that.

Reply:
Good catch, I'm thinking of reimplementing this either as `selectExpr` or converting it into a panadsUDF.

Reply:
I have switched this to use native spark functionality rather than udfs.